feat: change mls boolean switcher to protocol select (FS-700)

PatrykBuniX opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-700" title="FS-700" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />FS-700</a>  [Web] New Design for protocol selection for new Group 
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Change mls boolean switcher to protocol select

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

thisisamir98 committed:
runfix: Avoid duplicated system messages for MLS groups (#13422)

* fix: remove creationMessage in case we initiated the event

* chore: solution didnt work

* feat: filter duplicated MemberMessages in conversations

* feat: add guard

Co-authored-by: Adrian Weiß <github@weissadrian.com>

aweiss-dev committed:
Merge branch 'dev' into edge

aweiss-dev committed:
fix: compiler bug

aweiss-dev committed:
Merge branch 'dev' into edge

PatrykBuniX committed:
feat: use select for protocol selection

PatrykBuniX committed:
runfix: fix group creation modal's styles

PatrykBuniX committed:
Merge branch 'dev' into feat/FS-700-mls-select

PatrykBuniX committed:
refactor: reuse theme object

przemvs commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -70,17 +72,37 @@ enum GroupCreationModalState {
+  const initialProtocol: ProtocolOption = {
+    label: t('modalCreateGroupProtocolSelect.proteus'),
+    value: ConversationProtocol.PROTEUS,
+  };
```
Use protocolOptions.find(protocol => protocol.value === ConversationProtocol.PROTEUS) :)

przemvs commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -225,187 +247,205 @@ const GroupCreationModal: React.FC<GroupCreationModalProps> = ({
+            <div className="group-creation__list" ref={setScrollbarRef}>
+              {getContacts().length > 0 && (
+                <UserSearchableList
+                  users={getContacts()}
```
Move getContacts() to variable ;)

PatrykBuniX replied:
used `useMemo`

przemvs commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -70,17 +72,37 @@ enum GroupCreationModalState {
+}
+
+function isProtocolOption(option?: Option | null): option is ProtocolOption {
+  const protocols = Object.values(ConversationProtocol) as string[];
```
I think you can move it outside isProtocolOption. And move this typeguard to scripts/guards/<e.g: Protocol.ts>.

PatrykBuniX replied:
does it makes sense? It's being used only once in one file?

przemvs replied:
Maybe you will use it later, good to have all typeguards in one place. Check there structure:
https://github.com/wireapp/wire-webapp/pull/13373/files#diff-ff5e3819c325e9820395a1d0f449dfbe21a346b505bad93db411c7ad9926ccc6

PatrykBuniX replied:
ok

przemvs reviewed (changes requested):
Few changes ;)

V-Gira commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -53,6 +54,7 @@ import {
 import useEffectRef from 'Util/useEffectRef';
 import {useFadingScrollbar} from '../../../ui/fadingScrollbar';
 import {Config} from '../../../Config';
+import {theme} from '../../../page/MainContent/MainContent';
```
I'm making a note of this. The theme is going to live in the ui-kit soon. I'll make sure to update this

PatrykBuniX replied:
Sounds good, thanks!

przemvs commented on src/script/components/ModalComponent.tsx:
```diff
@@ -150,7 +152,7 @@ const ModalComponent: React.FC<ModalComponentProps> = ({
           tabIndex={-1}
           onKeyDown={noop}
-          css={hasVisibleClass ? ModalContentVisibleStyles : ModalContentStyles}
+          css={{...(hasVisibleClass ? ModalContentVisibleStyles : ModalContentStyles), ...wrapperCSS}}
```
I'm thinking about it, you replace by wrapperCSS previous css styles. It should be before or after this modalcontent styles ?

PatrykBuniX replied:
It comes after the default styles so we can overwrite it if we want to.

PatrykBuniX committed:
refactor: apply cr suggestions

PatrykBuniX committed:
Merge branch 'edge' into feat/FS-700-mls-select

tlebon committed:
chore: merge dev

przemvs committed:
fix(input-bar): open modal after error (#13458)

V-Gira committed:
runfix: restore theme_id.light in auth page (#13460)

dependabot[bot] committed:
chore(deps-dev): Bump @betterer/cli from 5.3.7 to 5.4.0 (#13461)

Bumps [@betterer/cli](https://github.com/phenomnomnominal/betterer) from 5.3.7 to 5.4.0.
- [Release notes](https://github.com/phenomnomnominal/betterer/releases)
- [Changelog](https://github.com/phenomnomnominal/betterer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/phenomnomnominal/betterer/compare/v5.3.7...v5.4.0)

---
updated-dependencies:
- dependency-name: "@betterer/cli"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump eslint-plugin-jsdoc from 39.3.4 to 39.3.6 (#13464)

Bumps [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) from 39.3.4 to 39.3.6.
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v39.3.4...v39.3.6)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump simple-git from 3.11.0 to 3.12.0 (#13462)

Bumps [simple-git](https://github.com/steveukx/git-js/tree/HEAD/simple-git) from 3.11.0 to 3.12.0.
- [Release notes](https://github.com/steveukx/git-js/releases)
- [Changelog](https://github.com/steveukx/git-js/blob/main/simple-git/CHANGELOG.md)
- [Commits](https://github.com/steveukx/git-js/commits/simple-git@3.12.0/simple-git)

---
updated-dependencies:
- dependency-name: simple-git
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @typescript-eslint/parser from 5.32.0 to 5.33.0 (#13466)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.32.0 to 5.33.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.33.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/core from 29.1.3 to 29.1.7 (#13468)

Bumps [@wireapp/core](https://github.com/wireapp/wire-web-packages) from 29.1.3 to 29.1.7.
- [Release notes](https://github.com/wireapp/wire-web-packages/releases)
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/core@29.1.3...@wireapp/core@29.1.7)

---
updated-dependencies:
- dependency-name: "@wireapp/core"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump eslint from 8.21.0 to 8.22.0 (#13475)

Bumps [eslint](https://github.com/eslint/eslint) from 8.21.0 to 8.22.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.21.0...v8.22.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-hot-middleware from 2.25.1 to 2.25.2 (#13474)

Bumps [webpack-hot-middleware](https://github.com/webpack-contrib/webpack-hot-middleware) from 2.25.1 to 2.25.2.
- [Release notes](https://github.com/webpack-contrib/webpack-hot-middleware/releases)
- [Commits](https://github.com/webpack-contrib/webpack-hot-middleware/compare/v2.25.1...v2.25.2)

---
updated-dependencies:
- dependency-name: webpack-hot-middleware
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump eslint-plugin-jest from 26.7.0 to 26.8.2 (#13469)

Bumps [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest) from 26.7.0 to 26.8.2.
- [Release notes](https://github.com/jest-community/eslint-plugin-jest/releases)
- [Changelog](https://github.com/jest-community/eslint-plugin-jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jest-community/eslint-plugin-jest/compare/v26.7.0...v26.8.2)

---
updated-dependencies:
- dependency-name: eslint-plugin-jest
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @betterer/typescript from 5.3.7 to 5.4.0 (#13470)

Bumps [@betterer/typescript](https://github.com/phenomnomnominal/betterer) from 5.3.7 to 5.4.0.
- [Release notes](https://github.com/phenomnomnominal/betterer/releases)
- [Changelog](https://github.com/phenomnomnominal/betterer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/phenomnomnominal/betterer/compare/v5.3.7...v5.4.0)

---
updated-dependencies:
- dependency-name: "@betterer/typescript"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @typescript-eslint/eslint-plugin (#13463)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.32.0 to 5.33.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.33.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

arjita-mitra committed:
fix: Device button accessibility(ACC-242) (#13478)

chat participants profile info are not executable(ACC-235)
Group admin and description not accessibile

V-Gira committed:
chore: bump avs to 8.2.8 acc-243 (#13479)

* chore: bump avs to 8.2.8

dependabot[bot] committed:
chore(deps): Bump wire-web-config-default-staging in /app-config (#13365)

Bumps [wire-web-config-default-staging](https://github.com/wireapp/wire-web-config-default) from v0.31.1 to v0.31.2.
- [Release notes](https://github.com/wireapp/wire-web-config-default/releases)
- [Commits](https://github.com/wireapp/wire-web-config-default/compare/v0.31.1...fe2a0d37c9750332b727d91bc86af16d5a05145a)

---
updated-dependencies:
- dependency-name: wire-web-config-default-staging
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

otto-the-bot committed:
chore: Update translations (#13442)

tlebon committed:
Merge branch 'dev' into feat/FS-700-mls-select

V-Gira commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -17,11 +17,12 @@
 import cx from 'classnames';
 import {amplify} from 'amplify';
 import {WebAppEvents} from '@wireapp/webapp-events';
+import {Select, StyledApp} from '@wireapp/react-ui-kit';
```
```suggestion
import {Select, StyledApp, THEME_ID} from '@wireapp/react-ui-kit';
```
The change to the ui-kit is live on dev

V-Gira commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -53,6 +54,8 @@ import {
 import useEffectRef from 'Util/useEffectRef';
 import {useFadingScrollbar} from '../../../ui/fadingScrollbar';
 import {Config} from '../../../Config';
+import {theme} from '../../../page/MainContent/MainContent';
```
```suggestion
```

V-Gira commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -225,187 +235,205 @@ const GroupCreationModal: React.FC<GroupCreationModalProps> = ({
-                conversationRepository={conversationRepository}
-                noUnderline
-              />
+    <StyledApp theme={theme}>
```
```suggestion
    <StyledApp themeId={THEME_ID.DEFAULT}>
```

PatrykBuniX replied:
Shouldn't it be `themes[THEME_ID.DEFAULT]` ? `themes` imported from ui-kit also

V-Gira replied:
you could do `theme={themes[THEME_ID.DEFAULT]}`
but we export the `THEME_ID` enum and we have a `themeId` prop

PatrykBuniX replied:
right, didn't spot the propname change here 😄

V-Gira reviewed (commented):
I made a couple of suggestion so the component is in line with the Theme update on dev

PatrykBuniX committed:
refactor: use themeId prop in modal theme wrapper

V-Gira commented on src/script/page/MainContent/MainContent.tsx:
```diff
@@ -51,6 +51,47 @@ const Animated: React.FC<{children: React.ReactNode}> = ({children, ...rest}) =>
+    primaryColor: 'var(--accent-color)',
+  },
+};
+
```
I somehow missed that massive green blob :D. The theme lives in ui-kit now, you can remove it from MainContent

PatrykBuniX replied:
ok, removed it

PatrykBuniX committed:
refactor: remove theme object from main content file

przemvs commented on src/page/template/panel/group-participant-user.htm:
```diff
@@ -15,16 +15,22 @@
+          <button class="panel__action-item" 
+          data-bind="click: onToggleAdmin, attr: {'aria-label': t('accessibility.conversationDetailsActionGroupAdminLabel')}" 
+          data-uie-name="toggle-admin" type="button"
+          aria-pressed={isAdmin()}>
```
It shouldn't be in data-bind ?

PatrykBuniX replied:
i don't think so, this is not a change I made, probably came from dev merge

przemvs commented on src/script/components/Modals/GroupCreation/GroupCreationModal.tsx:
```diff
@@ -225,187 +234,205 @@ const GroupCreationModal: React.FC<GroupCreationModalProps> = ({
+                    handleEnterDown(event, clickOnNext);
+                  }}
+                  value={groupName}
+                  isError={nameError.length > 0}
```
Move nameError.length > 0 to variable and replace other places where you're using.

PatrykBuniX replied:
its being used 2 times only, isn't it more readable this way? What should the variable be called? 🤔

przemvs replied:
For >= 2 usages is better to have variable. E.g. hasNameError :)

PatrykBuniX replied:
ok

przemvs commented on src/script/guards/ProtocolOption.ts:
```diff
@@ -0,0 +1,31 @@
+/*
```
I'm thinking about name. Maybe should be Protocol instead of ProtocolOption, because we will store all Protocol guards there.

przemvs commented on src/script/guards/ProtocolOption.ts:
```diff
@@ -0,0 +1,31 @@
+
+export function isProtocolOption(option?: Option | null): option is ProtocolOption {
+  const protocols = Object.values(ConversationProtocol) as string[];
+  return !!option && typeof option.value === 'string' && protocols.includes(option.value);
```
It doesn't work?

```suggestion
  return typeof option?.value === 'string' && protocols.includes(option.value);
```

przemvs commented on src/script/guards/ProtocolOption.ts:
```diff
@@ -0,0 +1,31 @@
+  value: ConversationProtocol;
+}
+
+export function isProtocolOption(option?: Option | null): option is ProtocolOption {
```
I'll go with arrow fn, in other guards we use arrow fn. Also parameter should be any | undefined | null :) Like in other cases.

PatrykBuniX replied:
<img width="281" alt="Screenshot 2022-08-22 at 15 44 15" src="https://user-images.githubusercontent.com/45733298/185937463-b8a6b250-2d4a-4877-aa40-d22111643ce0.png">
but `any | undefined | null` is equal to just `any` 🤔

PatrykBuniX replied:
`undefined` comes from `?` idk if we really want to allows to pass anything (`any`) here... Its being used in select options so it has to always be at least `Option | null`, WDYT?

przemvs replied:
Hmm... what if you will use it in different case ? And it get not an Option ?

PatrykBuniX replied:
so I guess just `option: any` is enough?

przemvs reviewed (changes requested):
Few changes. :)

PatrykBuniX committed:
refactor: apply CR suggestions
